src/JSInterop/Microsoft.JSInterop/src/JSInteropTask.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using System.Text.Json;
+
+namespace Microsoft.JSInterop;
+
+internal sealed class JSInteropTask<TResult> : IJSInteropTask
```

Comment:
I guess this means we're doing a few extra allocations per JS interop call, right?

Not suggesting that's a flaw in the design, just want to check my assumptions. If this is a requirement to work properly with AOT and gives people to improve the perf further using source-generated serializers, that would outweigh a few allocations.